internal/pkg/template/artifactpath/artifactpath.go

```diff
@@ -30,6 +31,12 @@ func Addons(key string, content []byte) string {
 	return path.Join(s3ArtifactDirName, s3ArtifactAddonsDirName, key, fmt.Sprintf("%x.yml", sha256.Sum256(content)))
 }
 
+// AddonAsset returns the path to store an addon asset file.
+// Example: manual/addons/addonKey/assets/hash.
+func AddonAsset(addonKey, hash string) string {
```

Comment:
nit: just to make it clear the key is a workload name
```suggestion
// Example: manual/addons/frontend/assets/ba7816bf8f01.
func AddonAsset(key, hash string) string {
```